src/core/unit.c

```diff
@@ -4538,10 +4542,22 @@ int unit_patch_contexts(Unit *u) {
                         if (r < 0)
                                 return r;
 
+                        r = cgroup_add_device_allow(cc, "block-device-mapper", "rwm");
+                        if (r < 0)
+                                return r;
+
                         /* Make sure "block-loop" can be resolved, i.e. make sure "loop" shows up in /proc/devices */
                         r = unit_add_two_dependencies_by_name(u, UNIT_AFTER, UNIT_WANTS, "modprobe@loop.service", true, UNIT_DEPENDENCY_FILE);
                         if (r < 0)
                                 return r;
+
+                        /* Same for mapper and verity */
+                        r = unit_add_two_dependencies_by_name(u, UNIT_AFTER, UNIT_WANTS, "modprobe@dm_mod.service", true, UNIT_DEPENDENCY_FILE);
+                        if (r < 0)
+                                return r;
+                        r = unit_add_two_dependencies_by_name(u, UNIT_AFTER, UNIT_WANTS, "modprobe@dm_verity.service", true, UNIT_DEPENDENCY_FILE);
+                        if (r < 0)
+                                return r;
```

Comment:
grmbl. This sucks, since there are plenty cases where Verity isn't used in the images, and we shouldn't needlessly pull in this modules if we likely won't need them....

But I have no better idea, so maybe leave it as it is.

(But maybe use FOREACH_STRING here too)

Reply:
Yeah it's not great. Unfortunately libdevmapper doesn't load the module itself, if it is not already, as far as I can remember.